src/intl/IntlDateTimeFormat.cpp

```diff
@@ -275,34 +268,77 @@ UTF16StringDataNonGCStd updateHourCycleInPatternDueToHourCycle(const UTF16String
 }
 
 
-void IntlDateTimeFormat::initialize(ExecutionState& state, Object* dateTimeFormat, Value locales, Value options)
+static String* icuFieldTypeToPartName(ExecutionState& state, int32_t fieldName)
 {
-    // If dateTimeFormat has an [[initializedIntlObject]] internal property with value true, throw a TypeError exception.
-    AtomicString initializedIntlObject = state.context()->staticStrings().lazyInitializedIntlObject();
-    if (dateTimeFormat->hasInternalSlot() && dateTimeFormat->internalSlot()->hasOwnProperty(state, ObjectPropertyName(state, ObjectStructurePropertyName(initializedIntlObject)))) {
-        ErrorObject::throwBuiltinError(state, ErrorObject::TypeError, "Cannot initialize Intl Object twice");
+    if (fieldName == -1) {
```

Comment:
What does `-1` mean? and why not use `< 0`?

Reply:
It's special value for ICU